### DISCUSSION TITLE
Alpine package

### DISCUSSION THREAD

**Discussion by uichleba** [rapier1/openssh-portable#82] [General]
Hi,
I just finished packing HPNSSH for Alpine Linux and would like to push merge request into a alpine repository. My question is if You would like to be a package maintainer or if You're Ok with that I would take care of it?
If anything I can always change ownership of that package to anyone else. Just let me know @rapier1.

Thank You.

**Comment by uichleba** [comment]
Btw. here is merge request - https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/64746 with `APKBUILD` if you want to check it.

**Comment by uichleba** [comment]
https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/64746#top merge request is closed and merged into a `/testing` aports repository.

**Reply by uichleba** [reply]
If somebody would be interested at testing `hpnssh` Alpine package, You may install it with:
`apk add hpnssh --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing/`

**Comment by rapier1** [comment]
Hey, sorry for the delay in replying. I've been working on an updated release (18.4.1) for later today. I'll be taking a look at this shortly. I just didn't want you to think I was ignoring you.

**Comment by rapier1** [comment]
As an aside, I would love it if you would handle package maintenance for HPN-SSH on Alpine. I'd like to know a bit more about the process and how you'd be handling updates and the like if you are okay with that. You may also want to get on the hpnssh-community mailing list. You can find that at https://lists.psc.edu/mailman/listinfo/hpnssh-community. It's a low volume list - usually it's just me doing release announcements. 

Chris

**Reply by uichleba** [reply]
Hi,
Sure I can handle package maintenance for Alpine - not a problem.
I subscribed to community mailing list when new version will be release I'll take take of packing.
As for a packing process, I spend some time with it. Started with docker where I use Alpine image where I compile `hpnssh` from master. After that I comment compiling and just prepared environment for packing - so all needed libs was already imorted.
After I had a image I run interactive container where I copied `APKBUILD` file and just testing it with a `abuild -r` cmd.
Then it's just matter of forking `aports` repository and make a merge request.
OR you choose much better approach that I switched to after that - [`alpkg`](https://github.com/orhun/alpkg).

**Comment by rapier1** [comment]
Hey there, I just installed an Alipne VM and installed the hpnssh APK. Seems to work as expected. Just a couple of questions, do you run "make tests" after compilation and ensure that it passes all of the regression tests? Do you package the tests as well (nto a big deal if you don't. Some distros do and some don't). Did you have to remove/elide any major features? 

Also, what do you think the chances are of this getting into the main APK repository? 

Lastly, 18.4.1 is now available. This fixes a potential issue with one of the buffers exploding in some circumstances leading to disconnects.

**Reply by uichleba** [reply]
Hi Chris,
- I made another MR with updated version `18.4.1` - https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/65575, so It should be updated.
- I do not run `make tests`. I maybe made a mistake because trying it before throw me an error that no `tests` in make. I did that and most of the tests are passing, but something has failed. Where would you like me to pass you a log? Should I put it here or maybe create a new Issue with that ?
- I did not have to remove any major feature or at least I don't know about any ..
- `hnpssh` package is in `edge/testing` official Alpine repository already. I do not know about progression from `testing` into `main`/`community` in the next LTS Alpine release. `hpnssh` is my first package into Alpine so I can't tell, but when I do I will let You know.

Lukas

**Comment by uichleba** [comment]
Hi,
So I would like to answer to Your question @rapier1 if and when `hpnssh` will move into `community` and into a official release.
All informations could be found here: https://gitlab.alpinelinux.org/alpine/aports/-/tree/master in README.md file.
TLDR:
- after it's verified that it's working it will be moved from `testing` ASAP
- I will get notification after 6.months if it's not moved from `testing`
- IDK when or who and how it gets verified that it's working (because I'm testing it and it's working just fine)
